builtin.py

```diff
@@ -11,13 +12,17 @@ def summary(self):
 # Base node classes
 
 class Node:
-    def __init__(self, title, desc):
+    def __init__(self, title: str, desc: str, connections: list[Node] = []):
         self.title = title
         self.desc = desc
+        self.connections = connections
```

Comment:
I like the idea behind this change, and I see how you tried to keep the API intact while implementing it. My only issue with it is that it interferes with the constructor parameters for the class, which was by design something I attempted to avoid. 

That being said, if we can get it working with another PR without many changes to the file parser, then maybe it's worth it!

Reply:
I changed the constructor parameters back to how they were. No changes to the parser should be needed.

Reply:
Perfect, that fixes everything! This is ready to merge now.